### PR TITLE
fwupd: 1.1.2 -> 1.2.1

### PR DESCRIPTION
**Pull request by dtzWill**
https://github.com/hughsie/fwupd/blob/1.1.3/NEWS

Upgrade to 1.1.2 broke things but has known fix that should be landing soon--
this bump needs same fix so test appropriately. :)

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

### CODE REVIEW

**Comment by dtzWill** [comment]
cc #48427  (the PR mentioned above).

**Comment by dtzWill** [comment]
@GrahamcOfBorg build fwupd
@GrahamcOfBorg test fwupd

(not sure if tests pass currently, but let's see...)

**Comment by dtzWill** [comment]
libxmlb 1.1.5 is out, do we want to include it as well?

**Comment by jtojnar** [comment]
Yup.

**Comment by jtojnar** [comment]
We need to wait for https://github.com/NixOS/nixpkgs/pull/50876

**Comment by dtzWill** [comment]
Well FWIW bumping to 0.1.5 (and adding a python3 dependency, apparently) and rebasing onto staging -- produces a bunch of build errors that didn't seem obvious.  My internet is being wonky or I'd be linking a branch/commit as well as logs, but anyway just a heads-up I suppose.

Oh hey managed to get the logs to gist anyway: https://gist.github.com/dtzWill/6412a806c9993ef01e1b4be4e20c938d

**Comment by jtojnar** [comment]
Needs `docbook_xsl docbook_xml_dtd_43` in `nativeBuildInputs`.

**Comment by dtzWill** [comment]
(still needs meson bump that's on staging, as earlier mentioned)

**Comment by jtojnar** [comment]
@GrahamcOfBorg build fwupd

**Comment by jtojnar** [comment]
I guess we should squash the commits.

**Comment by worldofpeace** [comment]
@jtojnar @dtzWill 

I've resolved the conflicts, corrected the commit msg, etc.
I hope you don't mind :sweat_smile:

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 1]
```diff
@@ -28,15 +30,16 @@ in stdenv.mkDerivation {
...
     libgudev colord gpgme libuuid gnutls glib-networking efivar json-glib umockdev
-    bash-completion
+    bash-completion cairo freetype fontconfig pango
   ];
 
   LC_ALL = "en_US.UTF-8"; # For po/make-images
 
   patches = [
     ./fix-paths.patch
+    ./add-option-for-installation-sysconfdir.patch
```
Comment (worldofpeace) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix]: I brought this patch in tree because, IIRC, the url before wouldn't have been stable.

**Comment by jtojnar** [comment]
@GrahamcOfBorg test fwupd

**Comment by worldofpeace** [comment]
Failed locally for me with
```
Installing test firmware...
Decompressing…         [***************************************]
failed to add public keys to gpgme-v1.12.0 keyring: failed to load /etc/pki/fwupd/GPG-KEY-Linux-Foundation-Metadata: No such file or directory
FAIL: fwupd/fwupdmgr.test (Child process exited with code 1)
error: Attempt to end element "line" with "nest" tag at /nix/store/vpm0bbgi88jj52sq9kfnpif5jrmqxcr9-nixos-test-driver/lib/perl5/site_perl/Logger.pm line 53.
Attempt to end element "line" with "nest" tag at /nix/store/vpm0bbgi88jj52sq9kfnpif5jrmqxcr9-nixos-test-driver/lib/perl5/site_perl/Logger.pm line 53.
cleaning up
killing machine (pid 597)
Attempt to end element "line" with "logfile" tag at /nix/store/vpm0bbgi88jj52sq9kfnpif5jrmqxcr9-nixos-test-driver/lib/perl5/site_perl/Logger.pm line 27.
END failed--call queue aborted, <__ANONIO__> line 98.
vde_switch: EOF on stdin, cleaning up and exiting
vde_switch: Could not remove ctl dir '/build/vde1.ctl': Directory not empty
builder for '/nix/store/3ac86f46d96zyjpsvqa5nyq0adsij4jl-vm-test-run-fwupd.drv' failed with exit code 22
error: build of '/nix/store/3ac86f46d96zyjpsvqa5nyq0adsij4jl-vm-test-run-fwupd.drv' failed
```

**Comment by worldofpeace** [comment]
Test succeeds locally, and I double checked `filesInstalledToEtc`.

File: pkgs/development/libraries/libxmlb/default.nix [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+  src = fetchFromGitHub {
+    owner = "hughsie";
+    repo = "libxmlb";
+    rev = version;
+    sha256 = "037j9fwkzsy3765gl2grkrmbxrfs67wlai213qbgsa5xn6fb8y68";
+  };
+
+  nativeBuildInputs = [ meson ninja pkgconfig gobject-introspection gtk-doc shared-mime-info docbook_xsl docbook_xml_dtd_43 ];
+
+  buildInputs = [ glib libuuid python3 ];
```
Comment (worldofpeace) [inline pkgs/development/libraries/libxmlb/default.nix]: I think `python3` here is probably only needed during the build.

Comment (jtojnar) [reply pkgs/development/libraries/libxmlb/default.nix]: Sounds good.

**Comment by jtojnar** [comment]
Something to keep in mind for next update: https://github.com/hughsie/fwupd/pull/865

**Comment by worldofpeace** [comment]
I think everything looks good. Lets go for it :+1:

**Pull request merged**